browser-extension/mv2/src/modules/EventActions.js

```diff
@@ -109,32 +141,67 @@ EventActions.sendExtensionEvents = async () => {
 };
 
 EventActions.writeEventsToLocalStorage = async () => {
-  const createBatch = (eventsArray) => {
-    const batchId = RQ.commonUtils.generateUUID();
+  if (EventActions.eventsCount > EventActions.EVENTS_LIMIT) {
```

Comment:
If I understand correctly, batches are created only when App loads and before that it will keep pushing events to an in-memory queue.
How about checking the limit while writing events to the queue and dropping all entries from the queue if the limit exceeds? This way, you don't need to maintain a size counter and keep in sync with every operation.

Reply:
No, batches are not created on app load. Batches are created in 10s interval whenever there are events in the queue.

Reply:
@nafees87n Yeah, missed that. You may ignore the comment then.